hs_core/management/commands/regenerate_xml_files_composite_resource.py

```diff
@@ -0,0 +1,73 @@
+"""Generate bag files composite resources. Regeneration of bag files is needed to get the
+resource level xml files corrected as well as to generate aggregation level
+xml files which currently does not exist for any of the composite resources created prior to
+composite resource phase-2 implementation
+
+* By default, prints errors on stdout.
+* Optional argument --log: logs output to system log.
+"""
+
+from django.core.management.base import BaseCommand
+
+from hs_core.hydroshare import create_bag_files
+from hs_composite_resource.models import CompositeResource
+
+
+class Command(BaseCommand):
+    help = "Generate bag files for specified composite resources"
+
+    def add_arguments(self, parser):
+
+        # a list of resource id's, or none to check all resources
+        parser.add_argument('resource_ids', nargs='*', type=str)
+
+        # Named (optional) arguments
+        parser.add_argument(
+            '--log',
+            action='store_true',  # True for presence, False for absence
+            dest='log',           # value is options['log']
+            help='log errors to system log',
+        )
+
+    def handle(self, *args, **options):
+        resource_counter_success = 0
+        resource_counter = 0
+
+        # inner function
+        def generate_bag_files(resource):
+            metadata_status = {}
+            for aggregation in resource.logical_files:
+                is_dirty = aggregation.metadata.is_dirty
+                metadata_status[aggregation.id] = is_dirty
+                # this is needed to generate aggregation level xml files
+                aggregation.metadata.is_dirty = True
+                aggregation.metadata.save()
+            print("> GENERATING BAG FILES FOR COMPOSITE RESOURCE:{}".format(resource.short_id))
+            create_bag_files(resource)
+            print(">> GENERATED BAG FILES FOR COMPOSITE RESOURCE:{}".format(resource.short_id))
+            # re-set metadata status
+            for aggregation in resource.logical_files:
+                aggregation.metadata.is_dirty = metadata_status[aggregation.id]
```

Comment:
I'm not following this.  Shouldn't all aggregation metadata be clean at this point?  The lines before set them all to dirty and then regenerate the bag files.

Reply:
Hope my related comment above helps you here.

Reply:
I'm still not following, the loop before each aggregation metadata is set to dirty and saved.  Are they not all clean at this point?  Why reset the status?

Reply:
I can remove the code that resets metadata dirty flag which is not going to cause that big of a problem. This resetting is necessary only for netcdf and timeseries aggregation. In order to create aggregation level xml files, metadata for the aggregation needs to be dirty state. Each aggregation type sets the dirty flag to false after the xml docs are created except for nectcdf and timeseries. Now if we don't reset the dirty flag as part of the regenerating the aggregations then the metadata will be left in dirty state for all netcdf and timeseries aggregations. This will result in user seeing message in UI for these aggregations that metadata is out of sync with content files (which is not true) and user will get the option to write the metadata to the content file.  I guess this message or the option to update the content file is not a big deal.